evm-precompiles/erc1155/src/lib.rs

```diff
@@ -213,16 +213,17 @@ where
 	}
 
 	fn is_precompile(&self, address: H160, _remaining_gas: u64) -> IsPrecompileResult {
+		let extra_cost = RuntimeHelper::<Runtime>::db_read_gas_cost();
 		if let Some(collection_id) =
 			Runtime::evm_id_to_runtime_id(Address(address), ERC1155_PRECOMPILE_ADDRESS_PREFIX)
 		{
 			// Check whether the collection exists
 			IsPrecompileResult::Answer {
 				is_precompile: pallet_sft::Pallet::<Runtime>::collection_exists(collection_id),
-				extra_cost: 0, // TODO: account gas for above storage read?
+				extra_cost
 			}
 		} else {
-			IsPrecompileResult::Answer { is_precompile: false, extra_cost: 0 }
+			IsPrecompileResult::Answer { is_precompile: false, extra_cost }
```

Comment:
read cost occurs at `pallet_sft::Pallet::<Runtime>::collection_exists(collection_id)`? hence the else path should not have any extra costs
```suggestion
			IsPrecompileResult::Answer { is_precompile: false, extra_cost: 0 }
```